main.tf

```diff
@@ -81,14 +81,23 @@ resource "aws_wafv2_web_acl" "main" {
 # WAFv2 web acl association with ALB
 #####
 resource "aws_wafv2_web_acl_association" "main" {
-  count = var.enabled && var.create_alb_association ? 1 : 0
+  count = var.enabled && var.create_alb_association && var.alb_arn != "" ? 1 : 0
```

Comment:
With this change we are getting an error:
```
Error: Invalid count argument

  on ../../main.tf line 84, in resource "aws_wafv2_web_acl_association" "main":
  84:   count = var.enabled && var.create_alb_association && var.alb_arn != "" ? 1 : 0

The "count" value depends on resource attributes that cannot be determined
until apply, so Terraform cannot predict how many instances will be created.
To work around this, use the -target argument to first apply only the
resources that the count depends on.
```
This is a common thing with count unfortunately. This would only work if you provide a specific string. When you reference a module it will fail. You can replicate that behaviour by running core example.

Reply:
it should be alright now, sorry for the delay